### DISCUSSION TITLE
Modes which could really use bookmark support

### DISCUSSION THREAD

**Discussion by jdtsmith** [alphapapa/activities.el#74] [Ideas]
Thought it would be nice to compile a list of high-priority modes for which bookmark support would be useful. I think @oantolin was looking at this too.  

Current list:

- [ ]  `Custom-mode`. 
- [X] `magit`
- [ ]  `compilation-mode`
- [ ] `grep`/`occur`
- [ ]  `xref`

**Comment by alphapapa** [comment]
Great idea.  I will nominate:

-  `compilation-mode`
- `xref--xref-buffer-mode` (which comes from using `project-find-regexp`): It would be nice to be able to bookmark one of these buffers, showing all of the matches for a regexp in a project and/or directory, and come back to it later.

**Comment by oantolin** [comment]
Here's a link to that [Reddit post](https://www.reddit.com/r/emacs/s/ZFFjuKfDu6) of mine you mentioned.

**Reply by jdtsmith** [reply]
Thanks. Maybe a meta-package which adds bookmark support for lots of different built-ins?

**Reply by alphapapa** [reply]
That wouldn't be a good idea, because it could become outdated as the related modes change.  Bookmark support is generally simple and should be added directly to the library in question.

**Comment by jdtsmith** [comment]
I would have put `magit` on this list, but turns out it's had bookmark support all along.  Just needed some [autoloads](https://github.com/magit/magit/discussions/5168).
